src/patch.js

```diff
@@ -32,15 +65,16 @@ export default class Patch {
 
     let startNode = this.iterator.insertSpliceBoundary(outputStart)
     startNode.isChangeStart = true
+    this.splayNode(startNode)
 
     let endNode = this.iterator.insertSpliceBoundary(outputOldEnd, startNode)
     endNode.isChangeEnd = true
+    this.splayNode(endNode)
     if (options && options.metadata) endNode.metadata = options.metadata
 
-    startNode.priority = -1
-    this.bubbleNodeUp(startNode)
-    endNode.priority = -2
-    this.bubbleNodeUp(endNode)
+    if (endNode.left !== startNode) {
```

Comment:
I am not sure we can, because we want `endNode` to be at the top of the tree. When splaying that node, however, a `zig-zig` could take place and, as a result, `startNode` would become the left child of `endNode`'s initial parent (i.e. becoming the left child of `endNode`'s left child after the splay) and that's basically what that if clause is preventing.

![zig-zig](https://cloud.githubusercontent.com/assets/482957/12840153/dee11a28-cbe0-11e5-8269-3098b099b033.gif)

_(`10` would be the `startNode` in this case and `30` the `endNode`)_

Am I missing something? Perhaps we could write a comment to make that clearer?